Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp

```diff
@@ -138,7 +139,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               M_xface(i, j, k, 2) += dt * (-PhysConst::mu0 * mag_gamma_interp) * ( M_xface(i, j, k, 0) * Hy_eff - M_xface(i, j, k, 1) * Hx_eff)
                 + dt * Gil_damp * ( M_xface(i, j, k, 0) * ( M_xface(i, j, k, 2) * Hx_eff - M_xface(i, j, k, 0) * Hz_eff)
                 - M_xface(i, j, k, 1) * ( M_xface(i, j, k, 1) * Hz_eff - M_xface(i, j, k, 2) * Hy_eff));
- 
+
+              // temporary normalized magnitude of M_xface field at the fixed point
+              // re-investigate the way we do Ms interp, in case we encounter the case where Ms changes across two adjacent cells that you are doing interp
+              Real M_normalized = sqrt( pow(M_xface(i, j, k, 0),2.0) + pow(M_xface(i, j, k, 1),2.0) + pow(M_xface(i, j, k, 2),2.0) ) /
+                                  MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
+
+#ifdef AMREX_DEBUG
+              // check the normalized error
```

Comment:
Check with Jackie - do we want this to be a DEBUG feature only or always on?
I say always make this on - if people don't want this feature they can set M_normalized_error to a huge value.